hack/lib/test.sh

```diff
@@ -252,7 +254,7 @@ kube::test::describe_resource_events_assert() {
 
 # Compare sort-by resource name output with expected order specify in the last parameter
 kube::test::if_sort_by_has_correct_order() {
-  local array=($(echo "$1" |awk '{if(NR!=1) print $1}'))
+  IFS=" " read -r -a array <<< "$(echo "$1" |awk '{if(NR!=1) print $1}')"
```

Comment:
Fix for this:

```
In ./hack/lib/test.sh line 255:
  local array=($(echo "$1" |awk '{if(NR!=1) print $1}'))
               ^-- SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).
```

Reply:
I think this is what we want, but we also have `kube::util::read-array` for similar situations.

Reply:
Is your suggestion to source `hack/lib/util.sh` to make use of `kube::util::read-array`?

Reply:
this broke sort_by comparison, and the failure was masked by https://github.com/kubernetes/kubernetes/commit/313044abd7a23699b5d744c4157ca7392673d587#diff-7976a1974df5995ca28f32cc945e92acL135

Reply:
Thanks for the info on this @liggitt. Looks like you have the fix in #84871 already